Cobbduceus: Multiver Rebalance

ExcessiveUseOfCobblestone opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

`feedback` commit and recent commit changes:

Uniques required to inhibit increased
Toxin's unique score counts less (0.5 > 0.2)
Organ damage goes up to max 1 instead of down to min 0.3
Uniques affect all reagent purging, but still does bonus against tox
Uniques required to stop meds from purging increased and is relative to the toxin BTFO (5 > 7.5)


## About The Pull Request
Multiver has been rebalanced!

- Unique Non-medicines will now contribute to multiver's efficacy, albeit at a lower rate compared to medicine (half for toxins, tenth for everything else)
- Less toxin damage healing per unique because of this
- Lung damage now starts at 1.5 (2-0.5) and goes down in a linear fashion instead of diminishing returns. Caps at 0.5 damage, which is 3 unique meds.
- No longer specializes in purging alcohol (use antihol)
- **Multiver now completely disables toxins so long as you have a sufficient amount of unique reagents in your system**. Purging is **NOT** affected (see where I'm getting at?).

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

Multiver was not seen really anywhere outside of the antag arsenal. Given it's gimmick requires reagent dumps of valuable medicine that could be used elsewhere, it really inhibits (hehe) its usage in core play, especially when compared to Pacid or even Syrin.

Hopefully now it will see a lot more play and effectively establish itself as the C2 toxin purger. Advanced antag strategies could still use this as a way to delay effects of poisons at the cost of losing quite a bit to the purging effects, but you're probably better off just filling up on lexorin still 😏.

I would say this might hit kill syringes, but people are still racking up lung damage and are having to dump reagents they could be using later to save themselves to contain the kill syringe chem(s) so your chems aren't entirely in vain!

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl: Multiverbby
balance: Multiver does less toxin healing per unique (see above) and no longer specializes as an anti-booze.
balance: Multiver's lung damage no longer gives diminishing returns for uniques.
add: Multiver now scales with unique reagents, instead of unique medicines. Medicines are still the best at helping multiver though!
add: Multiver now shuts off toxins if there's enough uniques present.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

ExcessiveUseOfCobblestone committed:
ニャー

ExcessiveUseOfCobblestone committed:
うわああああああ

ExcessiveUseOfCobblestone committed:
cap this

ExcessiveUseOfCobblestone committed:
cringe math

Krysonism commented:
I don't like the most commonly available anti-toxin being a nearly perfect antidote to all realistic toxin mixes.

ExcessiveUseOfCobblestone commented:
I might revert the old change of going top down and instead make it do more damage the more purgepower it has. idk nightcode is danger

The chem alone does not act as a "nearly perfect antidote to all realistic toxin mixes", it requires you to pull your other resources that may not even be related to toxin (and come with their own downsides?).

There's also the ancient martial art of purging all chems at once with 1 quick trick that has 0 downsides but that's probably a bug :)

EgorDinamit commented:
Ah yes, you are speaking about 1u water + potassium.

Krysonism commented:
A bottle of water and the roundstart multiver pills makes you virtually immune to toxins. 

That is hilariously OP since it takes no effort.

Multiver is mostly fine as it is in my opinion, it does the job when you are hit by a syringe. The only problem is that it is outshined by pent acid, the solution is not to make multiver stronger than pent acid.

AdamElTablawy commented:
Do you come up with these ideas when on crack or is it just natural talent?

Obviously this will get merged anyway, because of course it will, but this is a horrible idea. It just stopping toxins immediately is just...lame.

If it was meant to be a chem purger, then its role of 'chemical purger but ith drawbacks' is already taken by fucking calomel. If it was meant to be a toxin healer, then it doesn't do that.

It's only possible utility is murder.

So, to compensate for this, you make it instead instantly stop a deathmix from affecting you. That's lame.

ExcessiveUseOfCobblestone commented:
It only does this if you’re continuing to uphold the threshold and unless there’s medicine in water that’s a pretty blatant misrepresentation of the code.

In regards to calo, these are not mutually exclusive and multiver is meant to be better considering organ damage surpasses toxin.

gitbirb commented:
couldn't it at least just SLOW metabolism of toxins instead of completely stop them?

ExcessiveUseOfCobblestone commented:
do you mean increase metabolism? You want it out as soon as possible.

Rohesie commented on code/modules/reagents/chemistry/holder.dm:
```diff
@@ -363,7 +363,7 @@
 		if(C && R)
 			if(C.reagent_check(R) != TRUE)
-				if(liverless && !R.self_consuming) //need to be metabolized
+				if((liverless && !R.self_consuming) || R.overrides_processing) //need to be metabolized or we're specifically overriding it's processing.
```
```suggestion
				if((liverless && !R.self_consuming) || R.overrides_processing) //need to be metabolized or we're specifically overriding its processing.
```
Minor typo

ExcessiveUseOfCobblestone committed:
feedback

Toxin Inhibition requires more Uniques

Toxins count less (fifth vs. half) but still more than the rest (tenth)

Purges All Non-self reagents more per unique (not just toxins), but still retains more toxin purging than non.

Med Non-purging now is dependent on define

ExcessiveUseOfCobblestone commented:
The same can be said with any C2 that does organ damage though. They're all weaponizable if you have enough.

With multiver you can drive that number down though.

@vpellen would make sense if toxins all did toxin damage, which they do not. Most toxins in the meta kill syringes aren't even focused on doing tox damage.

ExcessiveUseOfCobblestone committed:
Multiver scales up with uniques rather than down.

ExcessiveUseOfCobblestone commented:
It was an issue with old multiver too so I just did it increasing vs. going down with more uniques (rip multiver in the cryo tube).

ExcessiveUseOfCobblestone committed:
snake? SNAKKEEEE

ExcessiveUseOfCobblestone committed:
ok no more robustin testing, real code now...

Rohesie commented:
I suppose the concern does extend to any chems doing organ damage and allowing for kill syringes.
One possible solution is to increase the volume needed for both healing and killing, making the lethal syringes less likely.
All that said, I have no idea what mixes are viable currently, so this is more of an abstract concern than anything specific or a change request here.

ExcessiveUseOfCobblestone commented:
the secret is you're better off using dedicated poisons than trying to underhand ones that heal you in other aspects (at least for C2s).

The exception is/was multiver.

AtlasRules commented:
multiver is a better killer than healer, and this doesn't change shit balance-wise, it's gone from a better poison to ridiculously OP

ExcessiveUseOfCobblestone commented:
Can you explain which part is OP (or even unbalanced), preferably in a way that includes the PR'd drawbacks?

Can you explain which part makes it a better poison as well, compared to other poison mixes, preferably by pointing out which lines in the code make it so?

Just want to make sure we are on the same page with what's being proposed, since many have upvoted the "just add water" comment which wasn't even true when I first PR'd it.

MarioWizard119 commented:
While it’s no longer a poison, multiver in this state seems bad. Even as its supposed role as a purger, it’s a piss in the ocean compared to the other meds.

You’d need 2 unique meds (along with multiver) for it to be on par with Syriniver+ Musiver
9 for it to be on par with pentetic
And 12 for it to be on par with calomel

Considering you need 7 other unique meds for it to stop purging meds, and you can’t chem stack it with other toxin meds, since they purge each other (Except seiver), it really makes it counter intuitive and a pain in the ass for luke warm results. Your solutions to get multiver to decent effectiveness are either to fill it with junk, which would take more effort than it’s worth, at that point you’d be better off making pentetic, or fill it with other meds, which there aren’t enough salines and granibitaluris in the world to make it decent, and if you use any other cat 2s would result in unnecessary organ damage. Compared to the other early game cat 2s, like libital and aiuri, multiver is extremely underwhelming as a medicine.

If you want to make multiver viable, scrap the gimmick, make it consistent, make it reliable, make it effective, make it good in the early game instead of having to jump through a million hoops just to get it to not suck at its job.

ExcessiveUseOfCobblestone commented:
I'll entertain that, this is the only chem left that continuously is a pain.
